test/IRGen/weak.sil

```diff
@@ -28,9 +28,9 @@ public struct A {
 }
 
 // size 8
-// flags 0x130007 == 1245191 (non-POD, non-inline, non-bitwise-takable)
+// flags 0x170007 == 1507335 (non-POD, non-inline, non-bitwise-takable, has extra inhabitants)
 // stride 8
-// CHECK: @"$S4weak1AVWV" = {{.*}} i8* inttoptr (i64 8 to i8*), i8* inttoptr (i64 1245191 to i8*), i8* inttoptr (i64 8 to i8*)]
+// CHECK: @"$S4weak1AVWV" = {{.*}} i8* inttoptr (i64 8 to i8*), i8* inttoptr (i64 1507335 to i8*), i8* inttoptr (i64 8 to i8*)
```

Comment:
This still shouldn't be changing.  Weak references don't have extra inhabitants because we're reserving everything about the reference for the use of the runtime.

Reply:
Well, that's on the "bonus" commit, which is being dropped. Thanks for clarifying that the runtime reserves all weak bit patterns. I'm adjusting the pull request and updating the tests.

Do we have a test that verifies that the extra inhabitant count is the same for the scenario you mentioned (`Foo<T:class>` to `Foo<ConcreteNativeClass>`)? If we do, I haven't tripped over it yet.